src/main/java/io/quarkus/bot/CancelWorkflowOnClosedPullRequest.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.bot;
+
+import io.quarkiverse.githubapp.event.PullRequest;
+import io.quarkus.bot.config.QuarkusBotConfig;
+import io.quarkus.bot.workflow.WorkflowConstants;
+import org.jboss.logging.Logger;
+import org.kohsuke.github.GHEventPayload;
+import org.kohsuke.github.GHPullRequest;
+import org.kohsuke.github.GHWorkflowRun;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+class CancelWorkflowOnClosedPullRequest {
+    private static final Logger LOG = Logger.getLogger(CancelWorkflowOnClosedPullRequest.class);
+
+    @Inject
+    QuarkusBotConfig quarkusBotConfig;
+
+    public void onClose(@PullRequest.Closed GHEventPayload.PullRequest pullRequestPayload) throws IOException {
+
+        GHPullRequest pullRequest = pullRequestPayload.getPullRequest();
+
+        List<GHWorkflowRun> ghWorkflowRuns = pullRequest.getRepository()
+                .queryWorkflowRuns()
+                .branch(pullRequest.getHead().getRef())
+                .list()
+                .toList()
+                .stream()
+                .filter(workflowRun -> WorkflowConstants.QUARKUS_CI_WORKFLOW_NAME.equals(workflowRun.getName()) &&
```

Comment:
I think we should also cancel the documentation job.

Reply:
Looking at the documentation job name, I can see that `Quarkus CI` actually match two jobs (the incremental one and the "normal" one) do you think it could cause issue?

Reply:
No, the normal one has been disabled.